### ISSUE TITLE
Lora finetune need more GPU memory than full finetune Hunyuan video?

### ISSUE THREAD

**Issue by AI-follow** [hao-ai-lab/FastVideo#167]
### Environment

- FastVideo version: 1.2.0
- Platform: Linux-5.15.0-97-generic-x86_64-with-glibc2.35
- Python version: 3.10.0
- PyTorch version: 2.5.1+cu121 (GPU)
- Transformers version: 4.46.1
- Accelerate version: 1.0.1
- PEFT version: 0.13.2
- GPU type: NVIDIA H200
- Bitsandbytes version: 0.45.0

### Describe the bug

Why Lora finetune need more GPU memory than full finetune Hunyuan video?  During my test, lora finetune needs about 130G memory per-GPU, but full finetune needs around 118G memory per-GPU with the same following parameters ;

torchrun --nnodes 1 --nproc_per_node 8 --master_port 29703 \
    train.py \
    --seed 42 \
    --pretrained_model_name_or_path data/FastHunyuan \
    --dit_model_name_or_path data/FastHunyuan/hunyuan-video-t2v-720p/transformers/mp_rank_00_model_states.pt\
    --model_type "hunyuan" \
    --cache_dir data/.cache \
    --data_json_path data/HD-Mixkit-Finetune-Hunyuan/videos2caption.json \
    --validation_prompt_dir data/HD-Mixkit-Finetune-Hunyuan/validation \
    --gradient_checkpointing \
    --train_batch_size=1 \
    --num_latent_t 32 \
    --sp_size 1 \
    --train_sp_batch_size 1 \
    --dataloader_num_workers 2 \
    --gradient_accumulation_steps=1 \
    --max_train_steps=2000 \
    --learning_rate=1e-5 \
    --mixed_precision=bf16 \
    --checkpointing_steps=20000 \
    --validation_steps 10000 \
    --validation_sampling_steps 50 \
    --checkpoints_total_limit 3 \
    --allow_tf32 \
    --ema_start_step 0 \
    --cfg 0.0 \
    --ema_decay 0.999 \
    --log_validation \
    --output_dir=data/outputs/HSH-Taylor-Finetune-Hunyuan_2 \
    --tracker_project_name HSH-Taylor-Finetune-Hunyuan \
    --num_frames 125 \
    --num_height 720 \
    --num_width 1280 \
    --shift 7 \
    --validation_guidance_scale "1.0" \

### Reproduction

torchrun --nnodes 1 --nproc_per_node 8 --master_port 29703 \
    train.py \
    --seed 42 \
    --pretrained_model_name_or_path data/FastHunyuan \
    --dit_model_name_or_path data/FastHunyuan/hunyuan-video-t2v-720p/transformers/mp_rank_00_model_states.pt\
    --model_type "hunyuan" \
    --cache_dir data/.cache \
    --data_json_path data/HD-Mixkit-Finetune-Hunyuan/videos2caption.json \
    --validation_prompt_dir data/HD-Mixkit-Finetune-Hunyuan/validation \
    --gradient_checkpointing \
    --train_batch_size=1 \
    --num_latent_t 32 \
    --sp_size 1 \
    --train_sp_batch_size 1 \
    --dataloader_num_workers 2 \
    --gradient_accumulation_steps=1 \
    --max_train_steps=2000 \
    --learning_rate=1e-5 \
    --mixed_precision=bf16 \
    --checkpointing_steps=20000 \
    --validation_steps 10000 \
    --validation_sampling_steps 50 \
    --checkpoints_total_limit 3 \
    --allow_tf32 \
    --ema_start_step 0 \
    --cfg 0.0 \
    --ema_decay 0.999 \
    --log_validation \
    --output_dir=data/outputs/HSH-Taylor-Finetune-Hunyuan_2 \
    --tracker_project_name HSH-Taylor-Finetune-Hunyuan \
    --num_frames 125 \
    --num_height 720 \
    --num_width 1280 \
    --shift 7 \
    --validation_guidance_scale "1.0" \

**Comment by BrianChen1129** [comment]
could you provide the complete LoRA finetuning script? It should enable --use_lora and appropriate --lora_rank/lora_alpha settings.  Also, you can set sp_size to 8 if finetuning on 8GPUs

**Comment by ghost** [comment]
> could you provide the complete LoRA finetuning script? It should enable --use_lora and appropriate --lora_rank/lora_alpha settings. Also, you can set sp_size to 8 if finetuning on 8GPUs

We use the default lora finetuning script like following 
torchrun --nnodes 1 --nproc_per_node 8 --master_port 29903 \
    fastvideo/train.py \
    --seed 1024 \
    --pretrained_model_name_or_path data/hunyuan_diffusers \
    --model_type hunyuan_hf \
    --cache_dir data/.cache \
    --data_json_path data/HD-Mixkit-Finetune-Hunyuan/videos2caption.json \
    --validation_prompt_dir data/HD-Mixkit-Finetune-Hunyuan/validation \
    --gradient_checkpointing \
    --train_batch_size 1 \
    --num_latent_t 32 \
    --sp_size 1 \
    --train_sp_batch_size 1 \
    --dataloader_num_workers 2 \
    --gradient_accumulation_steps 1 \
    --max_train_steps 600000 \
    --learning_rate 8e-5 \
    --mixed_precision bf16 \
    --checkpointing_steps 500 \
    --validation_steps 100 \
    --validation_sampling_steps 50 \
    --checkpoints_total_limit 3 \
    --allow_tf32 \
    --ema_start_step 0 \
    --cfg 0.0 \
    --ema_decay 0.999 \
    --log_validation \
    --output_dir data/outputs/Hunyuan-lora-finetuning-HD-Mixkit \
    --tracker_project_name Hunyuan-lora-finetuning-Black-Myth-Wukong \
    --num_frames 125 \
    --validation_guidance_scale "1.0" \
    --shift 7 \
    --use_lora \
    --lora_rank 32 \
    --lora_alpha 32

**Comment by BrianChen1129** [comment]
could you try `--sp_size 8` ?

**Comment by BrianChen1129** [comment]
I tried LoRA fine-tuning using different `--sp_size` = 2,4,8. And to test `--sp_size=1`, I reduced both `--num_latent_t` and `--num_frames` . Across all configurations, the results consistently showed that LoRA fine-tuning require less memory compared to full fine-tuning.  Could you check your setting again? Also, if you are planning to finetune your own model, we recommend set  `--sp_size=8` when finetune on 8 GPUs
